clang/lib/CIR/Dialect/IR/CIRTypes.cpp

```diff
@@ -615,6 +615,200 @@ IntType::verify(llvm::function_ref<mlir::InFlightDiagnostic()> emitError,
   return mlir::success();
 }
 
+//===----------------------------------------------------------------------===//
+// FloatType Definitions
+//===----------------------------------------------------------------------===//
+
+Type cir::FloatType::parse(mlir::AsmParser &parser) {
+  auto *ctx = parser.getBuilder().getContext();
+  Format format;
+
+  if (parser.parseLess())
+    return {};
+
+  // Fetch floating-point format.
+  llvm::StringRef formatStr;
+  if (parser.parseKeyword(&formatStr))
+    return {};
+  if (formatStr == "bin16")
+    format = FF_Binary16;
+  else if (formatStr == "bin32")
+    format = FF_Binary32;
+  else if (formatStr == "bin64")
+    format = FF_Binary64;
+  else if (formatStr == "bin64e")
+    format = FF_Binary64E;
+  else if (formatStr == "bin128")
+    format = FF_Binary128;
+  else if (formatStr == "bf16")
+    format = FF_BFloat16;
+  else
+    return {};
+
+  if (parser.parseGreater())
+    return {};
+
+  return cir::FloatType::get(ctx, format);
+}
+
+void cir::FloatType::print(mlir::AsmPrinter &printer) const {
+  const char *formatStr;
+  switch (getFormat()) {
+  case FF_Binary16:
+    formatStr = "bin16";
+    break;
+
+  case FF_Binary32:
+    formatStr = "bin32";
+    break;
+
+  case FF_Binary64:
+    formatStr = "bin64";
+    break;
+
+  case FF_Binary64E:
+    formatStr = "bin64e";
+    break;
+
+  case FF_Binary128:
+    formatStr = "bin128";
+    break;
+
+  case FF_BFloat16:
+    formatStr = "bf16";
+    break;
+
+  default:
+    llvm_unreachable("unknown floating-point format");
+  }
+
+  printer << '<' << formatStr << '>';
+}
+
+uint64_t cir::FloatType::getBitWidth() const {
+  switch (getFormat()) {
+  case FF_Binary16:
+  case FF_BFloat16:
+    return 16;
+
+  case FF_Binary32:
+    return 32;
+
+  case FF_Binary64:
+    return 64;
+
+  case FF_Binary64E:
+    return 80;
+
+  case FF_Binary128:
+    return 128;
+
+  default:
+    llvm_unreachable("unknown floating-point format");
+  }
+}
+
+std::string cir::FloatType::getAlias() const {
+  switch (getFormat()) {
+  case FF_Binary16:
+    return "f16";
+
+  case FF_Binary32:
+    return "f32";
```

Comment:
The aliases should look more like C/C++, like `float` and `double` (in the future `long_double` or `ldouble` for `long double` in the future, etc).